### PR TITLE
Fix broken links found in link checker

### PR DESCRIPTION
**Pull request by clay-aptos**
### Description

Used w3 link checker and Dr.  Link check to find broken links

### Test Plan

Manually repaired and tested them.

- - -
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/4665)

### CODE REVIEW

**Review by rajkaramchedu — APPROVED** [review]
I think we no longer use the nodes/index.md file after last week's overhaul. Can you check that and delete the file (I think I am quite sure we don't use the file, but my eyes could be deceiving me). Rest looks good.

**Comment by clay-aptos** [comment]
> I think we no longer use the nodes/index.md file after last week's overhaul. Can you check that and delete the file (I think I am quite sure we don't use the file, but my eyes could be deceiving me). Rest looks good.

Thanks, Raj! Confirmed the only reference was commented out and so deleted the file here:

```
claymurphy@MacBook-Air ~ % grep -r "nodes/index" aptos-core 
aptos-core/developer-docs-site/sidebars.js:      link: { type: "doc", id: "nodes/index" },
```

**Comment by rajkaramchedu** [comment]
Alert: Docs build failed. I think a redirect in the docusaurus.js file from the deleted doc to `/nodes/validator-node/validators` might fix the issue. Strictly speaking this new target is only for validator node, but that should do for now, methinks.

**Comment by rajkaramchedu** [comment]
In any case, click on the Details in the Checks section for the deploy preview and you will see a bunch of broken link errors from Docusaurus.

**Comment by clay-aptos** [comment]
> In any case, click on the Details in the Checks section for the deploy preview and you will see a bunch of broken link errors from Docusaurus.

Yeah, this is confusing. I am surmising the file you had me delete (`nodes/index.md`) somehow gets transformed into this file, which most certainly is still in use and accessible from the Nodes drop-down menu at the top of the site:
https://aptos.dev/nodes/nodes-index

And on that page you will find all of the 404s I originally fixed in `nodes/index.md` broken there. Perhaps we shouldn't delete that file after all? Please advise. Thanks!

**Comment by clay-aptos** [comment]
> > In any case, click on the Details in the Checks section for the deploy preview and you will see a bunch of broken link errors from Docusaurus.
> 
> Yeah, this is confusing. I am surmising the file you had me delete (`nodes/index.md`) somehow gets transformed into this file, which most certainly is still in use and accessible from the Nodes drop-down menu at the top of the site: https://aptos.dev/nodes/nodes-index
> 
> And on that page you will find all of the 404s I originally fixed in `nodes/index.md` broken there. Perhaps we shouldn't delete that file after all? Please advise. Thanks!

Raj and I fixed this by removing the top nav link to the old Nodes index.

File: developer-docs-site/docs/guides/move-guides/move-on-aptos.md [hunk 1]
```diff
@@ -38,7 +38,7 @@ The Aptos Move adapter features include:
 The Aptos framework ships with many useful libraries:
 * A [Token standard](https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/framework/aptos-token/sources/token.move) that makes it possible to create NFTs and other rich tokens without publishing a smart contract
 * A [Coin standard](https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/framework/aptos-framework/sources/coin.move) that makes it possible to create type-safe Coins by publishing a trivial module
-* An [iterable Table](https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/framework/aptos-stdlib/sources/iterable_table.move) that allows for traversing all the entries within a table
+* An [iterable Table](https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/move-examples/data_structures/sources/iterable_table.move) that allows for traversing all the entries within a table
```
Comment (gregnazario) [inline developer-docs-site/docs/guides/move-guides/move-on-aptos.md:41]: I would remove `iterable-table` from this list as it's not part of the framework

**Review by gregnazario — APPROVED** [review]

File: developer-docs-site/docs/guides/move-guides/move-on-aptos.md (hunk 1 above)
Comment (rajkaramchedu) [reply developer-docs-site/docs/guides/move-guides/move-on-aptos.md]: Done.

**Pull request merged**